File: src/main/java/org/rak/doodles/Painting.java
package org.rak.doodles;

public class Painting {

    public static void useMeWithoutAClass(){

    }


    public boolean doThing(Pixel pixel) {
        float x = pixel.getX();
        pixel.setIndex(10);
        return false;
    }

    public String evaluate(int number, boolean specialCase) {
        if (number <= 0) {
            return "Bad Value";
        } else if (number == 1) {
            if (specialCase) {
                return "Special number one!";
            } else {
                return "Single";
            }
        } else if (number == 2) {
            return "Double";
        } else {
            return "Default";
        }
    }

}

File: src/main/kotlin/Painting.kt
fun doThing(pixel: Pixel) : Boolean {
    val x = pixel.x
    pixel.index = 10
    return false
}

fun transformList(list: List<String>, transformation: (String) -> String) : List<String> {
    return list.map { transformation(it) }
}

fun useMeWithoutAClass() {

}

fun evaluate(number: Int, specialCase: Boolean): String {
    return when {
        number <= 0 -> "Bad Value"
        number == 1 && specialCase -> "Special number one!"
        number == 1 -> "Single"
        number == 2 -> "Double"
        else -> "Default"
    }
}


fun evaluate(number: Int): String {
    return when (number) {
        1 -> "Single"
        2 -> "Double"
        else -> "Default"
    }
}